### DISCUSSION TITLE
NormalizeIntensity parameters cannot be scalars?

### DISCUSSION THREAD

**Discussion by agaldran** [Project-MONAI/MONAI#7420] [Q&A]
Hello!

So, I dont think I understand correctly this transform. If I use in my code:
```
tr_transforms = transformsCompose([
...
transforms.NormalizeIntensityd(keys=('img', ), channel_wise=True, nonzero=True)
...
])
```
then it works nicely, but when I try to pass it a mean and a std with which I would like my data to by normalized (I am assuming im = (im-mean)/std, then it crashes:
```
m, s = torch.tensor(347.14), torch.tensor(120.35)
tr_transforms = transformsCompose([
...
transforms.NormalizeIntensityd(keys=('img', ), subtrahend=m, divisor=s, channel_wise=True, nonzero=True)
...
])
```
so, what should i pass to this transform instead?

Thanks!

**Comment by KumoLiu** [comment]
Hi @agaldran,
Yes, you are right, if you pass `channel_wise=True`, it will normalize channel-wise.
https://github.com/Project-MONAI/MONAI/blob/6c9e49ee045c345ecd8d3518cb0d503ed02c61b7/monai/transforms/intensity/array.py#L894
https://github.com/Project-MONAI/MONAI/blob/6c9e49ee045c345ecd8d3518cb0d503ed02c61b7/monai/transforms/intensity/array.py#L907
https://github.com/Project-MONAI/MONAI/blob/6c9e49ee045c345ecd8d3518cb0d503ed02c61b7/monai/transforms/intensity/array.py#L885

Hope it helps, thanks.

**Comment by agaldran** [comment]
Hey thanks for the quick answer. So it seems that with `channel_wise=False` things work as expected. I was thinking that with only one channel scalars would have worked the same, but that's not true.

Does the normalization expect a list with as many items as channels? Would it work if I use a list of len 1 for both the mean and the std? Thanks!

tr_transforms = transformsCompose([
...
transforms.NormalizeIntensityd(keys=('img', ), subtrahend=[mean,], divisor=[std,], channel_wise=True, nonzero=True)
...
])

**Reply by KumoLiu** [reply]
The data should be channel first array.
https://github.com/Project-MONAI/MONAI/blob/6c9e49ee045c345ecd8d3518cb0d503ed02c61b7/monai/transforms/intensity/array.py#L890
